src/viam/robot/client.py

```diff
@@ -173,6 +196,52 @@ async def _refresh_every(self, interval: int):
             except Exception as e:
                 LOGGER.error("Failed to refresh status", exc_info=e)
 
+    async def _check_connection(self, check_every: int, reconnect_every: int):
+        while True:
+            if self._connected:
+                wait_time = check_every
+            else:
+                wait_time = reconnect_every
+
+            await asyncio.sleep(wait_time)
+
+            if not self._address:
```

Comment:
@njooma @dannenberg unfortunately we rely on knowing the address in order to reconnect, so a robot that was created directly through a call to `with_channel` would never be able to reconnect. I suspect this is fine since most use cases are going to be connecting via address, but wanted to flag it.